libres/tests/analysis/test_assert_update_viable.cpp

```diff
@@ -0,0 +1,143 @@
+                    continue;
+                }
+                state_map_iset(source_state_map, i, STATE_HAS_DATA);
+            }
+            REQUIRE(state_map_get_size(source_state_map) == 2);
+            REQUIRE(state_map_count_matching(source_state_map,
+                                             STATE_LOAD_FAILURE) == 1);
+
+            THEN("Assertion pass with realizations >= ensemble_size") {
+                REQUIRE(analysis::assert_update_viable(
+                    analysis_config, source_state_map, ensemble_size,
+                    updatestep));
+            }
+
+            ensemble_size += 1;
+            THEN("Assertion fail with realizations < ensemble_size") {
+                REQUIRE(!analysis::assert_update_viable(
+                    analysis_config, source_state_map, ensemble_size,
+                    updatestep));
+            }
+        }
+
+        WHEN("0 > realizations > min_realizations") {
+            // analysis_config_set_min_realisations is set as static function
+            // but it does not state why this is done since the struct is already
+            // declared locally and thus the function has a redundant use
+            // So for testing purposeses (the only outside use) its exposed,
+            // please chime your opinion on this change.
+            analysis_config_set_min_realisations(analysis_config, 2);
+            state_map_iset(source_state_map, 0, STATE_INITIALIZED);
+            state_map_iset(source_state_map, 0, STATE_HAS_DATA);
+            THEN("Assertion pass, realization >= ensemble_size") {
+                REQUIRE(analysis::assert_update_viable(
+                    analysis_config, source_state_map, ensemble_size,
+                    updatestep));
+            }
+            ensemble_size += 1;
+            THEN("Assertion fail, realization < ensemble_size") {
+                REQUIRE(!analysis::assert_update_viable(
+                    analysis_config, source_state_map, ensemble_size,
+                    updatestep));
+            }
+        }
+
+        WHEN("Analysis module flag which is not ANALYSIS_ITERABLE is set") {
+            local_updatestep_add_ministep(updatestep, ministep);
+            REQUIRE(local_updatestep_get_num_ministep(updatestep) == 2);
+            analysis_module_flag_enum analysis_module_flags[4] = {
+                ANALYSIS_NEED_ED, ANALYSIS_UPDATE_A, ANALYSIS_USE_A,
+                ANALYSIS_SCALE_DATA};
+            for (auto flag : analysis_module_flags) {
+                // TODO: set analysis_module_type of flag type to analysis_config
+                THEN("Assertion passes") {
+                    REQUIRE(!analysis::assert_update_viable( // change to True
+                        analysis_config, source_state_map, ensemble_size,
+                        updatestep));
+                }
+            }
+        }
+
+        // cleanup
+        local_ministep_free(ministep);
+        // analysis_module_free(analysis_module);
```

Comment:
Should the comments around this be removed?

Reply:
yeah we are still at the prototype, it's my lingering scaffolding